### PR TITLE
Add setting to disable standard hid braille

### PR DESCRIPTION
**Pull request by feerrenrut**
### Link to issue number:
None

### Summary of the issue:
Due to the late nature of changes to HID braille support, as a precaution add a work around in case users encounter issues.

### Description of how this pull request fixes the issue:
Allow users to prevent HID braille from being used.

### Testing strategy:
- Locally modified the config via the settings panel.
- Testing via RC

### Known issues with pull request:
None

### Change log entries:
```
Changes
- The new HID Braille protocol can be disabled via a setting in the advanced settings panel.
```

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English

### CODE REVIEW

**Comment by leonardder** [comment]
The documentation needs to be updated as well, since this is a new advanced setting.

File: source/gui/settingsDialogs.py [hunk 1]
```diff
@@ -2700,6 +2700,42 @@ def __init__(self, parent):
...
+			_("Yes"),
+			# Translators: Label for option in the 'Enable support for HID braille' combobox
+			# in the Advanced settings panel.
+			_("No"),
+		]
+
+		# Translators: This is the label for a checkbox in the
+		#  Advanced settings panel.
+		label = _("Enable support for HID braille")
+		self.supportHidBrailleCheckbox: wx.Choice = hidBrailleGroup.addLabeledControl(
```
Comment (CyrilleB79) [inline source/gui/settingsDialogs.py:2727]: Here and at other places, including comments:
Checkbox -> Combo

```suggestion
		self.supportHidBrailleCombo: wx.Choice = hidBrailleGroup.addLabeledControl(

```

**Comment by CyrilleB79** [comment]
As @leonardder writes, I think that the documentation needs to be updated; also do not forget context help. Even if documentation is not translated or if context help does not work for all languages, I think that it is better to have the English documentation up-to-date.

Regarding translation what is your plan:
1. Launch a new translation freeze allowing complete translation of UI and documentation
2. Do not allow translation before RC and 2021.3.1 release. If documentation and context help is updated in English anyway this leads to:
   * English NVDA fully translated
   * Non-English NVDA with a piece of UI in English, no documentation at all for this option and context help not working for this option.
3. Do not allow translation before RC and 2021.3.1 but paste manually the English paragraph in all the non-English documentations: This leads to:
   * English NVDA fully translated
   * Non-English NVDA with a piece of UI in English, a piece of documentation in English and context help working.
   * Risk related to a non-conventional way to update the documentation (merge conflicts in SVN, English parts remaining in 2022.1, etc.)

**Comment by feerrenrut** [comment]
Given this is advanced setting, and only likely to be changed in response to a support request or some technical investigation. Also at this stage, I think it is an unlikely problem, I've not had any confirmed reports of ongoing issues. I'd prefer not to encounter the issues highlighted by @CyrilleB79 for updating the documentation. On balance I think it is better to address the user guide changes in 2022.1.

**Comment by CyrilleB79** [comment]
Just to be extra-clear regarding my position (and given @feerrenrut's last comment)

I really think that having an undocumented part of the UI is not a good practice. Indeed, in case of any braille issue, unofficial documentation will naturally appear on the mailing lists consisting by e-mails send by whoever saying "Try to change the HID parameter in the advanced settings panel".

On the contrary I personally prefer having an official documentation, even in English only, which can serve as a reference in case of discussions on the mailing lists. Regarding translations, this documentation will then be naturally translated in 2022.1.

This been said, the final decision is yours, NVAccess.

**Comment by feerrenrut** [comment]
@CyrilleB79 generally I agree with you, in this case there is not clear advice for this setting, we are approaching the limit for how long this point release can delayed. The documentation for this settings panel is, don't touch unless instructed to by support or know what you are doing. In this case it's actually very low risk, and the option is quite self documenting, there aren't many HID braille devices out there.

I'm going to push forward with an RC2

**Comment by CyrilleB79** [comment]
OK no problem. I understand your point.

**Comment by leonardder** [comment]
Honestly I'd still rather have the English docs updated as part of 2021.3.1. I don't think there's anything holding us back from that, right?

**Pull request merged**